server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java

```diff
@@ -23,12 +26,24 @@
 
     private List<SubRating> subRatings;
 
+    private static final float WEIGHT_SUM_EPSILON = 0.01f;
+
+    private static final float WEIGHT_SUM_EXPECTED = 1f;
+
     public RatingSystem(String id, String name, ObjectId ownerId, Integer size, List<SubRating> subRatings) {
         this.id = id;
         this.name = name;
         this.ownerId = ownerId;
         this.size = size;
         this.subRatings = subRatings;
+        Assert.isTrue(size >= 2, "Size must be at least 2");
+        Assert.isTrue(subRatings.size()>=1, "subRatings size must be at least 1");
+        Assert.isTrue(nonNull(name) && !name.isBlank(),"name cannot be blank");
+        Assert.isTrue(name.length()>=1 && name.length()<= 50, "name must be at least one character and less than 50 characters");
+        var weightSum = subRatings.stream().mapToDouble(SubRating::getWeight).sum();
+        var difference = WEIGHT_SUM_EXPECTED - weightSum;
```

Comment:
the floating point error could be in either direction so I think we should take the absolute value of the difference here? then we could also get rid of the >=0 check on line 45

Reply:
left a comment regarding the use of epsilon and use absolute value for difference.

Reply:
We want the tolerance to only be on the “less than” side we want to guarantee the overall scores will n it exceed the max range. Absolute value allows a weight 1.01 to be valid thus we could have a score that is greater than the range.